### ISSUE TITLE
Validator not reporting significant errors

### ISSUE THREAD

**Issue by peterthomsonou** [validator/validator#570]
Document checking completed. No errors or warnings to show.
Source

    <!DOCTYPE html>↩
    <head>↩
    <title>test</title>↩
     <body>↩
            <p>

Used the HTML parser.

Total execution time 2 milliseconds.

**Comment by unor** [comment]
The document

```html
<!DOCTYPE html>
<head>
<title>test</title>
<body>
<p>
```

is valid. 

The end and starts tags of the `html` element are optional in this context.  
The ends tags `</head>`, `</p>`, and `</body>` are optional in this context.

**Comment by sideshowbarker** [comment]
Closing per https://github.com/validator/validator/issues/570#issuecomment-343290719
